dask/array/rechunk.py

```diff
@@ -363,14 +363,15 @@ def find_merge_rechunk(old_chunks, new_chunks, block_size_limit):
 
     old_largest_width = [max(c) for c in old_chunks]
     new_largest_width = [max(c) for c in new_chunks]
+    old_largest_width_div = [max(1, w) for w in old_largest_width]
 
     graph_size_effect = {
         dim: len(nc) / len(oc)
         for dim, (oc, nc) in enumerate(zip(old_chunks, new_chunks))
     }
 
     block_size_effect = {
-        dim: new_largest_width[dim] / old_largest_width[dim]
+        dim: new_largest_width[dim] / old_largest_width_div[dim]
```

Comment:
Instead of creating the new variable, maybe we can do the following?

     new_largest_width[dim] / max(1, old_largest_width[dim])

or 

    new_largest_width[dim] / (old_largest_width[dim] or 1)

Reply:
Went with the second approach